core/client/fs/src/main/java/alluxio/client/block/stream/BlockInStream.java

```diff
@@ -115,9 +119,12 @@ public static BlockInStream createNettyBlockInStream(long blockId, long blockSiz
               blockSize, false, Protocol.RequestType.ALLUXIO_BLOCK, options));
       blockWorkerClient.accessBlock(blockId);
       return new BlockInStream(inStream, blockWorkerClient, closer, options);
-    } catch (RuntimeException e) {
-      CommonUtils.closeQuietly(closer);
-      throw e;
+    } catch (Throwable t) {
+      try {
+        throw closer.rethrow(t);
+      } finally {
+        closer.close();
+      }
```

Comment:
This code is repeated lots of places, it will be good to put them in a util. This is optional.